connmgr/connmanager.go

```diff
@@ -300,7 +310,20 @@ func (cm *ConnManager) Connect(c *ConnReq) {
 		atomic.StoreUint64(&c.id, atomic.AddUint64(&cm.connReqCount, 1))
 	}
 	log.Debugf("Attempting to connect to %v", c)
-	conn, err := cm.cfg.Dial(c.Addr.Network(), c.Addr.String())
+
+	var (
+		conn net.Conn
+		err  error
+	)
+
+	if cm.cfg.Dial != nil {
```

Comment:
It already been proven in `New` that both can't be `nil`, so this is going to needless reduce the test coverage percentage with impossible to hit conditions.

I'd rather see this do:

```Go
var con net.Conn
var err error
if cm.cfg.Dial != nil {
	conn, err = cm.cfg.Dial(c.Addr.Network(), c.Addr.String())
} else {
	conn, err = cm.cfg.DialAddr(c.Addr)
}
if err != nil {
	cm.requests <- handleFailed{c, err}
} else {
	cm.requests <- handleConnected{c, conn}
}
```